lib/react_test_utils.dart

```diff
@@ -0,0 +1,255 @@
+      _Simulate.callMethod('touchCancel', [element, eventData]);
+
+  static touchEnd(JsObject element, [Map eventData = const{}]) =>
+      _Simulate.callMethod('touchEnd', [element, eventData]);
+
+  static touchMove(JsObject element, [Map eventData = const{}]) =>
+      _Simulate.callMethod('touchMove', [element, eventData]);
+
+  static touchStart(JsObject element, [Map eventData = const{}]) =>
+      _Simulate.callMethod('touchStart', [element, eventData]);
+
+  static wheel(JsObject element, [Map eventData = const{}]) =>
+      _Simulate.callMethod('wheel', [element, eventData]);
+
+}
+
+/// Traverse all components in tree and accumulate all components where
+/// test(component) is true. This is not that useful on its own, but it's
+/// used as a primitive for other test utils
+///
+/// Included in Dart for completeness
+List findAllInRenderedTree(JsObject tree, ReactComponentFactory test) {
+  return _TestUtils.callMethod('findAllInRenderedTree', [tree, test]);
+}
+
+/// Like scryRenderedDOMComponentsWithClass() but expects there to be one
+/// result, and returns that one result, or throws exception if there is
+/// any other number of matches besides one.
+JsObject findRenderedDOMComponentWithClass(JsObject tree, String className) {
+  return _TestUtils.callMethod(
+      'findRenderedDOMComponentWithClass', [tree, className]);
+}
+
+/// Like scryRenderedDOMComponentsWithTag() but expects there to be one result,
+/// and returns that one result, or throws exception if there is any other
+/// number of matches besides one.
+JsObject findRenderedDOMComponentWithTag(JsObject tree, String tag) {
+  return _TestUtils.callMethod(
+      'findRenderedDOMComponentWithTag', [tree, tag]);
+}
+
+/// Same as scryRenderedComponentsWithType() but expects there to be one result
+/// and returns that one result, or throws exception if there is any other
+/// number of matches besides one.
+JsObject findRenderedComponentWithType(
+    JsObject tree, ReactComponentFactory componentType) {
+  return _TestUtils.callMethod(
+      'findRenderedComponentWithType', [tree, getFactory(componentType)]);
+}
+
+/// Returns true if element is a composite component.
+/// (created with React.createClass()).
+bool isCompositeComponent(JsObject instance) {
+  return _TestUtils.callMethod('isCompositeComponent', [instance]);
+}
+
+/// Returns true if instance is a composite component.
+/// (created with React.createClass()) whose type is of a React componentClass.
+bool isCompositeComponentWithType(
+    JsObject instance, ReactComponentFactory componentClass) {
+  return _TestUtils.callMethod(
+      'isCompositeComponentWithType', [instance, getFactory(componentClass)]);
+}
+
+/// Returns true if instance is a DOM component (such as a <div> or <span>).
+bool isDOMComponent(JsObject instance) {
+  return _TestUtils.callMethod('isDOMComponent', [instance]);
+}
+
+/// Returns true if element is any ReactElement.
+bool isElement(element) {
```

Comment:
Right, so since should only work on React elements (which are instances of [the `ReactElement` JS constructor/class](https://github.com/matthewbalvanz-wf/react-dart/blob/76bbed05ed4bcb3acc3dd4203e49ac6aceb0aa66/lib/react_with_addons.js#L9867), and thus of the Dart type `JsObject`) , and not actual DOM nodes (`Element`/`DOMElement` Dart types), shouldn't it be typed to `JsObject`?